vm-validator/src/vm_validator.rs

```diff
@@ -187,6 +187,13 @@ impl<S: StateView> ModuleCache for ValidationState<S> {
         Ok(if version == value_version {
             Some((module, version))
         } else {
+            // TODO: remove this once framework on mainnet is using the new option module
```

Comment:
Can we factor this out (replacement and deserialization) into a single method and reuse with the building in build() below? This duplication is exactly why we miss these things in the first place😓